.github/workflows/release.yml

```diff
@@ -8,33 +8,166 @@ on:
+      - name: Build Wasm
+        if: matrix.config.buildWasm
         run: make build
+      - name: Upload Wasm as GitHub artifact
+        if: matrix.config.buildWasm
+        uses: actions/upload-artifact@v3
+        with:
+          name: bart
+          path: target/wasm32-wasi/release/bartholomew.wasm
 
       - name: Build bart
-        run: make bart
+        run: cargo build --release --manifest-path=bart/Cargo.toml ${{ matrix.config.extraArgs }}
 
-      - name: generate checksums
+      - name: package release assets
+        if: runner.os != 'Windows'
+        shell: bash
         run: |
-          sha256sum target/wasm32-wasi/release/bartholomew.wasm > checksums-${{ env.RELEASE_VERSION }}.txt
-          sha256sum target/release/bart >> checksums-${{ env.RELEASE_VERSION }}.txt
+          mkdir _dist
+          cp ${{ matrix.config.targetDir }}/bart${{ matrix.config.extension }} _dist/
+          cd _dist
+          tar czf \
+            bart-${{ env.RELEASE_VERSION }}-${{ env.RUNNER_OS }}-${{ matrix.config.arch }}.tar.gz \
+            bart${{ matrix.config.extension }}
+
+      - name: package release assets
+        if: runner.os == 'Windows'
+        shell: bash
+        run: |
+          mkdir _dist
+          cp ${{ matrix.config.targetDir }}/bart${{ matrix.config.extension }} _dist/
+          cd _dist
+          7z a -tzip \
+            bart-${{ env.RELEASE_VERSION }}-${{ env.RUNNER_OS }}-${{ matrix.config.arch }}.zip \
+            bart${{ matrix.config.extension }}
+
+      - name: upload binary as GitHub artifact
+        if: runner.os != 'Windows'
+        uses: actions/upload-artifact@v3
+        with:
+          name: bart
+          path: _dist/bart-${{ env.RELEASE_VERSION }}-${{ env.RUNNER_OS }}-${{ matrix.config.arch }}.tar.gz
+
+      - name: upload binary as GitHub artifact
+        if: runner.os == 'Windows'
+        uses: actions/upload-artifact@v3
+        with:
+          name: bart
+          path: _dist/bart-${{ env.RELEASE_VERSION }}-${{ env.RUNNER_OS }}-${{ matrix.config.arch }}.zip
+
+  checksums:
+    name: generate release checksums
+    runs-on: ubuntu-latest
+    needs: assets
+    steps:
+      - name: set the release version (tag)
+        run: echo "RELEASE_VERSION=${GITHUB_REF/refs\/tags\//}" >> $GITHUB_ENV
+
+      - name: download release assets
+        uses: actions/download-artifact@v3
+        with:
+          name: bart
+
+      - name: generate checksums
+        run: sha256sum * > checksums-${{ env.RELEASE_VERSION }}.txt
 
       - name: Create release
         uses: softprops/action-gh-release@v1
         with:
           fail_on_unmatched_files: true
           generate_release_notes: true
           files: |
-            target/wasm32-wasi/release/bartholomew.wasm
-            target/release/bart
+            bartholomew.wasm
+            *.tar.gz
+            *.zip
             checksums-${{ env.RELEASE_VERSION }}.txt
+
+      - uses: actions/upload-artifact@v3
```

Comment:
Curious on if this step is needed, since we upload the checksums file when creating the release above?

Reply:
Oh I guess maybe not?  It adds the checksums file to the workflow artifact in case someone downloads that but I guess who cares.

Reply:
This was mostly a leftover from 10 commits of frantic flailing to get stuff to work